test/fw/doc/howtotest.rst

```diff
@@ -0,0 +1,175 @@
+How to write test suite/case
+============================
+
+Assumptions
+-----------
+
+The library and utility make several assumptions about the environment:
+
+- OS should be Unix or Linux
+- Password-less authentication should be setup on all systems
+- Required users are available (see ``pbs_config --check-ug``)
+- The PTL package should be installed on all systems
+- The file system layout should be same on all systems
+- PBS_CONF_FILE variable should be set on all systems
+
+Naming conventions, recommended practices, and guidelines
+---------------------------------------------------------
+
+Write the test in a filename prefixed by ``pbs_``
+followed by either the issue number, for example pbs_ABC.py or by a
+descriptive name, for example pbs_smoketest.py, (strongly
+favor the issue number format).
+
+Name the test class prefixed by ``Test`` followed by either the issue number from
+which it originates, for example TestABCD, if the test is not from a issue,
+consider using a descriptive short name, for example TestSmoke.
+
+Whenever applicable name the test cases using a descriptive name, noting that
+PTL will run the methods that start with ``test_``. The test cases running
+sequence is unordered (some claim that it is lexicographic ordering but it is
+best to not write your test suites based on such assumptions).
+
+Put every functionality that is common to all test cases in its own method,
+and consider adding it to the library or utility if it is a generic interface to
+PBS.
+
+PTL strongly follows PEP8 Python coding style. so please style your code to follow
+PEP8 Python codeing style. You can find PEP8 at http://www.python.org/dev/peps/pep-0008/
+
+Some info about PBSTestSuite
+----------------------------
+
```

Comment:
I have added the url redirecting to open confluence page which is already there for ptl test directory structure because this information may change.